src/webgpu/compat/api/validation/createBindGroupLayout_limits.spec.ts

```diff
@@ -0,0 +1,87 @@
+export const description = `
+Tests that, in compat mode, you can not create a bind group layout with with
+more than the max in stage limit even if the per stage limit is higher.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { range } from '../../../../common/util/util.js';
+import { RequiredLimitsTestMixin } from '../../../gpu_test.js';
+import { CompatibilityTest } from '../../compatibility_test.js';
+
+export const g = makeTestGroup(
+  RequiredLimitsTestMixin(CompatibilityTest, {
+    getRequiredLimits(adapter: GPUAdapter) {
+      return {
+        maxStorageBuffersInFragmentStage: adapter.limits.maxStorageBuffersInFragmentStage! / 2,
```

Comment:
Is there anything that skips the test in browsers that don't have this? (or skips if the adapter isn't a compat adapter)

Reply:
Good catch!

Updated the code in `gpu_test.ts` so it filters out any undefined adapter.limits.